src/test/java/org/mule/test/extension/file/common/api/util/ZonedDateTimeLowerOrEqualAssessmentTestCase.java

```diff
@@ -9,18 +9,19 @@
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
 
-import org.mule.extension.sftp.internal.util.LocalDateTimeLowerOrEqualAssessment;
+import org.mule.extension.sftp.internal.util.ZonedDateTimeLowerOrEqualAssessment;
 
-import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
 
 import org.junit.Test;
 
-public class LocalDateTimeLowerOrEqualAssessmentTestCase {
+public class ZonedDateTimeLowerOrEqualAssessmentTestCase {
 
-  private static final LocalDateTime LOWER_BOUND = LocalDateTime.of(1983, 4, 20, 21, 15);
-  private static final LocalDateTime UPPER_BOUND = LocalDateTime.of(2012, 3, 7, 18, 45);
+  private static final ZonedDateTime LOWER_BOUND = ZonedDateTime.of(1983, 4, 20, 21, 15, 0, 0, ZoneId.systemDefault()); // .of(1983, 4, 20, 21, 15);
+  private static final ZonedDateTime UPPER_BOUND = ZonedDateTime.of(2012, 3, 7, 18, 45, 0, 0, ZoneId.systemDefault()); // LocalDateTime.of(2012, 3, 7, 18, 45);
```

Comment:
@elcoya Shouldn't we add a test with two different dates with different zoneId representing the same instant? In order to validate the support of ZonedDateTime (otherwise, in a change to another data type for dates, nothing would fail if we stop supporting the time zone).

Reply:
Added ZonedDateTimeTimezoneDifferencesAssessmentTestCase.java for this purpose. And added a matching fix for it since there was a bug in the original code